spark/dl/src/main/scala/com/intel/analytics/bigdl/example/MLPipeline/OptimizerWrapper.scala

```diff
@@ -0,0 +1,131 @@
+import com.intel.analytics.bigdl.example.imageclassification.MlUtils.{testMean => _, testStd => _, _}
+import com.intel.analytics.bigdl.nn.ClassNLLCriterion
+import com.intel.analytics.bigdl.utils.{Engine, T}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+import org.apache.spark.SparkContext
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.mllib.linalg.DenseVector
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Row, SQLContext}
+
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * An example to show how to use DLEstimator fit to be compatible with ML Pipeline
+ * refer to README.md on how to run this example
+ */
+object OptimizerWrapper {
+
+  LoggerFilter.redirectSparkInfoLogs()
+  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+  def main(args: Array[String]): Unit = {
+    val inputs = Array[String]("Feature data", "Feature size", "Label data", "Label size")
+    trainParser.parse(args, new TrainParams()).map(param => {
+      val conf = Engine.createSparkConf()
+        .setAppName("MLPipeline Example")
+        .set("spark.task.maxFailures", "1")
+      val sc = new SparkContext(conf)
+      val sqLContext = new SQLContext(sc)
+      Engine.init
+
+      val trainData = Paths.get(param.folder, "/train-images-idx3-ubyte")
+      val trainLabel = Paths.get(param.folder, "/train-labels-idx1-ubyte")
+      val validationData = Paths.get(param.folder, "/t10k-images-idx3-ubyte")
+      val validationLabel = Paths.get(param.folder, "/t10k-labels-idx1-ubyte")
+
+      val model = LeNet5(classNum = 10)
+
+      val trainSet = DataSet.array(load(trainData, trainLabel), sc) ->
+        BytesToGreyImg(28, 28) -> GreyImgNormalizer(trainMean, trainStd) -> GreyImgToBatch(
+        param.batchSize)
+
+      val validationSet = DataSet.array(load(validationData, validationLabel), sc) ->
+        BytesToGreyImg(28, 28) -> GreyImgNormalizer(testMean, testStd) -> GreyImgToBatch(
+        param.batchSize)
+
+      val dataFrameRDD : RDD[MinibatchData[Float]] = trainSet.
+        asInstanceOf[DistributedDataSet[MiniBatch[Float]]].data(false).map(batch => {
+          val feature = batch.data
+          val label = batch.labels
+          val estimatorData = MinibatchData[Float](feature.storage().array(),
+            feature.size(), label.storage().array(), label.size())
+           estimatorData
+        })
+
+      var df : DataFrame = sqLContext.createDataFrame(dataFrameRDD).toDF(inputs : _*)
+
+      val criterion = ClassNLLCriterion[Float]()
+
+      var estimator = new DLEstimator[Float]().setInputCols(inputs)
+
+      val paramsTrans = ParamMap(
+        estimator.modelTrain -> model,
+        estimator.criterion -> criterion,
+        estimator.batchShape -> Array[Int](128, 28, 28))
+
+      estimator = estimator.copy(paramsTrans)
+
+      val pipeline = new Pipeline()
+
+      pipeline.setStages(Array(estimator))
```

Comment:
I don't think we need to use a Pipeline here. User will be confused and think this is the only way.